Automattic/jetpack
# AI Client: Separate AIControl UI from block logic and update messages

Fixes #36966

## Proposed changes:

* Separates the `AIControl` component into a base `AIControl`, a block-specific `BlockAIControl` component and an extension-specific `ExtensionAIControl`. This allows us to reuse the same base for the extensions input, which looks very similar but has different actions and behavior
* Updates the messages, adding a Storybook story
* Updates the Storybook story for the AIControl component and the two new components

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* `( cd projects/js-packages/storybook && pnpm run storybook:dev --no-open --debug-webpack )` to run Storybook
* Go to JS Packages > AI Client
* Check the Messages story with 3rftFw1oBgRusMWgfNTaVz-fi-4437_5481
* Check the AIControl, BlockAIControl and ExtensionAIControl stories
* Use the AI Assistant and AI Form blocks, check that the buttons and behaviors (loading, disabled, etc) are working as normal

At this point the new `ErrorMessage` will not replace the `errorComponent` for the block control. That will be done later to keep compatibility.
